system/CLI/GeneratorTrait.php

```diff
@@ -0,0 +1,318 @@
+
+	/**
+	 * Execute the command.
+	 *
+	 * @param array $params
+	 *
+	 * @return void
+	 */
+	protected function execute(array $params): void
+	{
+		$this->params = $params;
+
+		if ($this->getOption('namespace') === 'CodeIgniter')
+		{
+			CLI::write(lang('CLI.generator.usingCINamespace'), 'yellow');
+			CLI::newLine();
+
+			// @codeCoverageIgnoreStart
+			if (CLI::prompt('Are you sure you want to continue?', ['y', 'n'], 'required') === 'n')
+			{
+				CLI::newLine();
+				CLI::write(lang('CLI.cancelOperation'), 'light_yellow');
+				CLI::newLine();
+
+				return;
+			}
+			// @codeCoverageIgnoreEnd
+
+			CLI::newLine();
+		}
+
+		// Get the fully qualified class name from the input.
+		$class = $this->qualifyClassName();
+
+		// Get the file path from class name.
+		$path = $this->buildPath($class);
+
+		// Check if path is empty.
+		if (empty($path))
+		{
+			return;
+		}
+
+		// Overwriting files unknowingly is a serious annoyance, So we'll check if
+		// we are duplicating things, If 'force' option is not supplied, we bail.
+		if (! $this->getOption('force') && file_exists($path))
+		{
+			CLI::write(lang('CLI.generator.fileExist', [CLI::color(clean_path($path), 'red')]));
+			CLI::newLine();
+
+			return;
+		}
+
+		// Check if the directory to save the file is existing.
+		$dir = dirname($path);
+
+		if (! is_dir($dir))
+		{
+			mkdir($dir, 0755, true);
+		}
+
+		helper('filesystem');
+		
+		// Build the class based on the details we have, We'll be getting our file
+		// contents from the template, and then we'll do the necessary replacements.
+		if (! write_file($path, $this->buildContent($class)))
+		{
+			CLI::write(lang('CLI.generator.fileError', [clean_path($path)]), 'red');
+			CLI::newLine();
+
+			return;
+		}
+
+		if ($this->getOption('force') && file_exists($path))
+		{
+			CLI::write(lang('CLI.generator.fileOverwrite', [CLI::color(clean_path($path), 'yellow')]));
+			CLI::newLine();
+
+			return;
+		}
+
+		CLI::write(lang('CLI.generator.fileCreate', [CLI::color(clean_path($path), 'light_green')]));
+		CLI::newLine();
+	}
+
+	/**
+	 * Prepare options and do the necessary replacements.
+	 *
+	 * @param string $class
+	 *
+	 * @return string
+	 */
+	protected function prepare(string $class): string
+	{
+		return $this->parseTemplate($class);
+	}
+
+	/**
+	 * Change file basename before saving.
+	 * 
+	 * Useful for components where the file name has a date.
+	 *
+	 * @param string $filename
+	 *
+	 * @return string
+	 */
+	protected function basename(string $filename): string
+	{
+		return basename($filename);
+	}
+
+	/**
+	 * Parses the class name and checks if it is already qualified.
+	 *
+	 * @return string
+	 */
+	private function qualifyClassName(): string
+	{
+		// Gets the class name from input.
+		$class = $this->params[0] ?? CLI::getSegment(2) ?? '';
+
+		if (empty($class))
+		{
+			$class = CLI::prompt(lang('CLI.generator.className'), null, 'required'); // @codeCoverageIgnore
+			CLI::newLine();
+		}
```

Comment:
I'm not forcing it! I'm using your old code getClassName but just in 1 line

Reply:
Yes, I'm aware of that. `GeneratorCommand::getClassName()` does not have any prompts because class name was conceived to be optional. If name is needed, then child classes can just provide a prompt. For other generators, for example language, class names are not required. So it is better to have the trait treat class names as optional at first and have classes using it to make them required when needed.

Reply:
Actually, your solution can be acceptable too as most generators require class names. For the sake of other generators soon to be made which do not require class names, this prompt will become a problem. To get rid of this problem, I suggest extracting this part to its method getClassName(). This way if a future generator does not need class names with prompts it would just need to override that method, instead of overriding the whole qualifyClassName.

Reply:
Can we use a property something like . boolean $hasClassName = true, and change it status with method in a child class whether we need that?

Reply:
Ok. We can go with that. Is that method something like `setHasClassName(bool $hasClassName)`? If the primary purpose is to turn on/off the `$hasClassName` property, it can live with this trait. If you'll be implementing that boolean flag, I'll be expecting it in the `if` statement of the prompt.
```php
if ($this->hasClassName && is_null($class))
{
    // ...
}
```